packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/representationdata/repositories/IRepresentationDataRepository.java

```diff
@@ -30,6 +30,13 @@
  */
 @Repository
 public interface IRepresentationDataRepository extends ListPagingAndSortingRepository<RepresentationData, UUID>, ListCrudRepository<RepresentationData, UUID> {
+    @Query("""
+            SELECT id, project_id, target_object_id, description_id, label, kind, '' as content, created_on, last_modified_on, last_migration_performed, migration_version
+            FROM representation_data
+            WHERE id = :id
+            """)
+    Optional<RepresentationData> findMetadataById(UUID id);
```

Comment:
You'll be interested then by a small improvement I made in this very same file to read pretty much the same thing. We can read any kind of projections in a repository, in this case I think it would be a bit better for a couple of reasons:

- It does not create some data in memory which do not match the state of the database
- It makes it very clear that we did not fetch everything
- I don't think that we will keep this aggregate as one for a long time so it starts creating this separation between the two future aggregates of this bounded context

I'll update this part to make it very visible to all consumers of this bounded context that there are two entry points in the API to read data.

Reply:
Yes, see my comment above: https://github.com/eclipse-sirius/sirius-web/pull/3642#issuecomment-2188308592